### DISCUSSION TITLE
Prometheus Rules - Best Practices

### DISCUSSION THREAD

**Discussion by theduke** [cloudprober/cloudprober#509] [Q&A]
The format of the metrics provided by the Prometheus exporter doesn't seem to make it easy to write Prometheus rules for alertmanager.

Can someone provide examples and best practices for how to write those?

My thinking is something like:

In the last x minutes:
- if timeout counter increased, and success  counter has not increased => alert
- if validation failure counter increased, and success counter has not increased => alert

**Comment by manugarg** [comment]
@theduke Thanks for the question. You should use `success` and `total` metrics for alerting. I'd alert on something like:

Failure ratio over last one minute:
```
1 - (
        sum by(dst, probe) (increase(cloudprober_success[1m]))
          /
        sum by(dst, probe) (increase(cloudprober_total[1m]))
)
```

If you run 2 probes per minute, you'll see this ratio become 0.5 on first failure, and 1 on 2 consecutive failures.
